saleor/graphql/order/mutations/order_update.py

```diff
@@ -102,36 +106,64 @@ def get_instance(cls, info: ResolveInfo, **data):
         return any(
-            cleaned_input.get(field) is not None
+            field in modified_fields
             for field in ["shipping_address", "billing_address"]
         )
 
     @classmethod
-    def _save(cls, info: ResolveInfo, instance, cleaned_input, changed_fields):
-        update_fields = changed_fields
+    def _save(cls, instance_tracker: InstanceTracker) -> bool:
+        instance = cast(models.Order, instance_tracker.instance)
         with traced_atomic_transaction():
-            address_fields = save_addresses(instance, cleaned_input)
-            update_fields.extend(address_fields)
+            modified_foreign_fields = instance_tracker.get_foreign_modified_fields()
+            if "shipping_address" in modified_foreign_fields:
+                cls._save_address(
+                    instance.shipping_address,
+                    modified_foreign_fields["shipping_address"],
+                )
+
+            if "billing_address" in modified_foreign_fields:
+                cls._save_address(
+                    instance.billing_address,
+                    modified_foreign_fields["billing_address"],
+                )
+
+            modified_instance_fields = instance_tracker.get_modified_fields()
+            modified_instance_fields.extend(modified_foreign_fields)
+            if not modified_instance_fields:
+                return False
 
-            manager = get_plugin_manager_promise(info.context).get()
-            if cls.should_invalidate_prices(cleaned_input):
+            update_order_search_vector(instance, save=False)
+            modified_instance_fields.extend(["search_vector"])
+            if cls.should_invalidate_prices(modified_instance_fields):
                 invalidate_order_prices(instance)
-                update_fields.append("should_refresh_prices")
+                modified_instance_fields.extend(["should_refresh_prices"])
 
-            if update_fields:
-                instance.search_vector = FlatConcatSearchVector(
-                    *prepare_order_search_vector_value(instance)
-                )
-                update_fields.extend(["updated_at", "search_vector"])
+            cls._save_order_instance(instance, modified_instance_fields)
 
-                instance.save(update_fields=update_fields)
-                call_order_event(
-                    manager,
-                    WebhookEventAsyncType.ORDER_UPDATED,
-                    instance,
-                )
+            return True
+
+    @classmethod
+    def _save_order_instance(cls, instance, modified_instance_fields):
+        update_fields = ["updated_at"] + modified_instance_fields
+        instance.save(update_fields=update_fields)
+
+    @classmethod
+    def _save_address(cls, address_instance, modified_fields):
+        if address_instance._state.adding:
+            address_instance.save()
+        else:
+            address_instance.save(update_fields=modified_fields)
```

Comment:
👮 codecov
I believe it may be worth adding a testcase to make sure that we are indeed editing address correctly

Reply:
good catch, thanks